1/contrib/s2i/run

```diff
@@ -69,6 +60,12 @@ fi
 
 rm -rf /tmp/war
 
+echo "Processing Jenkins Kubernetes configuration (${CONFIG_PATH}) ..."
+envsubst < "${CONFIG_PATH}.tpl" > "${CONFIG_PATH}" && rm -f "${CONFIG_PATH}.tpl"
```

Comment:
this is tricky... if you supply your own `config.xml` then you will disable the support for k8s slave detection, right? I think the ideal solution here would be support both, `config.xml` and `config.xml.tpl`. If the later exists, we also do processing using the environment variables. And since users can define their own variables, they can also use this mechanism to parametrize their own config template.